src/inmanta/server/services/orchestrationservice.py

```diff
@@ -1221,26 +1226,17 @@ async def deploy(
             return attach_warnings(404, {"message": "No agent could be reached"}, warnings)
 
         await self.autostarted_agent_manager._ensure_scheduler(env.id)
-        allagents = [const.AGENT_SCHEDULER_ID]
 
-        present = set()
-        absent = set()
-        for agent in allagents:
-            client = self.agentmanager_service.get_agent_client(env.id, agent)
-            if client is not None:
-                incremental_deploy = agent_trigger_method is const.AgentTriggerMethod.push_incremental_deploy
-                self.add_background_task(client.trigger(env.id, agent, incremental_deploy))
-                present.add(agent)
-            else:
-                absent.add(agent)
+        client = self.agentmanager_service.get_agent_client(env.id, const.AGENT_SCHEDULER_ID)
+        if not client:
+            return attach_warnings(404, {"message": "Scheduler could not be reached"}, warnings)
 
-        if absent:
-            warnings.append("Could not reach agents named [%s]" % ",".join(sorted(list(absent))))
+        incremental_deploy = agent_trigger_method is const.AgentTriggerMethod.push_incremental_deploy
 
-        if not present:
-            return attach_warnings(404, {"message": "No agent could be reached"}, warnings)
+        for agent in agents_to_call:
+            self.add_background_task(client.trigger(env.id, agent, incremental_deploy))
```

Comment:
I'd prefer an `if agents_to_call` or `if all_agents` around the for loop here over the `None` in the list. But I'm OK with it as is if you prefer.